src/cryptoadvance/specter/user.py

```diff
@@ -308,8 +308,8 @@ def save_info(self, delete=False):
 
     def add_service(self, service_id: str, autosave: bool = True):
         """Add a Service to the User. Only updates what is listed in the sidebar."""
-        if service_id not in self.services:
-            self.services.append(service_id)
+        if service_id not in self._services:
+            self._services.append(service_id)
```

Comment:
Why is this necessary? Uh, oh, now i see the issue. This is necessary because of my bad implementation of the property:
```
    @property
    def services(self):
        if self._services:
            return self._services
        return []
```
It should be changed to:
```
    @property
    def services(self):
        if not self._services:
            self._services = []    
        return self._services
```
Do you agree?

Reply:
Either way the `User._services` seems a little inconsistent in that it has a "member"-like level of protection in the class but that isolation isn't really accomplishing anything; external code can directly manipulate the `user_obj.services` prop just as easily as if the var was declared as `User.services` without the prop.

But for the `add_service` method itself, it's within `User` which should be free to modify its own internal `self._services` "private" member whenever/wherever it needs to.

I think this all boils down to philosophy/semantics. Bottom line for me is that the left nav is showing the active services once again!

Reply:
Applied your patch to the `@property`.

Reply:
Yes, it's not java. The reason i converted it to a property was avoiding a "has not attribute _services" issue. However, not implemented well. It doesn't matter to me that much whether to use internal or external (even though if it 's possible to use external, it's better in order to avoid the above issue) but the main thing was that this revealed my bad naive implementation.
Good that it's fixed!